### PR TITLE
grafana- default datasource for thanos / update datasources logic

### PR DESCRIPTION
**Pull request by JesseGoodier**
## What does this PR change?

Change default Grafana datasource to thanos if enabled. 
Update Grafana datasource for Prometheus to allow for Grafana interval variables to work properly

## Does this PR rely on any other PRs?

- No

## How does this PR impact users? (This is the kind of thing that goes in release notes!)

Change default Grafana datasource to thanos if enabled. 
Update Grafana datasource for Prometheus to allow for Grafana interval variables to work properly

## How was this PR tested?

1. Thanos install
2. Prometheus install
3. BYO Prometheus install

## Have you made an update to documentation?

No changes needed

### CODE REVIEW

File: cost-analyzer/values.yaml [hunk 1]
```diff
@@ -935,15 +935,6 @@ grafana:
   rbac:
     # Manage the Grafana Pod Security Policy
     pspEnabled: false
-  datasources:
```
Comment (thomasvn) [inline cost-analyzer/values.yaml:938]: @jessegoodier Could you explain the reasoning behind removing `.Values.datasources`? I notice it's used here as well:

https://github.com/kubecost/cost-analyzer-helm-chart/blob/3a283b62bb914daf45bc4ead079eddbe0612e212/cost-analyzer/charts/grafana/templates/configmap.yaml#L24-L29

If we do decide to get rid of `datasources.yaml` we should also remove the above code snippet and test?

Comment (jessegoodier) [reply cost-analyzer/values.yaml:938]: I was struggling with the logic.

Can you take a look at the configmap logic now?

**Comment by srpomeroy** [comment]
We also discussed removing the data source defined in the main values file as it duplicates the data sources.
https://github.com/kubecost/cost-analyzer-helm-chart/pull/2302/files#diff-e232ea8a3d165b37f6212249d3718bb96e05b64a591f1fa36d77c9448c747527

**Comment by thomasvn** [comment]
@jessegoodier These changes look good to me! I've also run `helm template` against these changes, and all templates render properly when no additional `values.yaml` is passed.

**Review by thomasvn — APPROVED** [review]

**Comment by jessegoodier** [comment]
> We also discussed removing the data source defined in the main values file as it duplicates the data sources. https://github.com/kubecost/cost-analyzer-helm-chart/pull/2302/files#diff-e232ea8a3d165b37f6212249d3718bb96e05b64a591f1fa36d77c9448c747527

done! thanks for checking on that.

**Pull request merged**